Build: #7 Dockerfile

micheal-ndoh opened the pull request:
Created a docker file to automate and build docker image of the group generator program
Close #7

micheal-ndoh committed:
Created Dockerfile

Created a docker file to automate and build docker image of the group generator program

micheal-ndoh committed:
Create README.md

Created a README file for the group generator program which helps users understand the project scope, objectives and how to use the program with different options

micheal-ndoh committed:
Updated Dockerfile

Added a command to create HOME environment in the contain so that data can be saved and read by the program after run

marcjazz commented on README.md:
It wasn't mention on this issue that will add Readme file. 

Please commit this under a associated task #1

micheal-ndoh replied:
Ok Sir

micheal-ndoh committed:
Updated README.md

Update readme file that shows the user on how to run the docker file with a mounted volume to help store and persist data after the container closes

micheal-ndoh committed:
Updated README.md

Modified read me file on program usage option II so that users should use -v flag when running the docker container so as to store and access the application previous saved state

micheal-ndoh committed:
Delete README.md

marcjazz commented on Dockerfile:
```diff
@@ -0,0 +1,24 @@
+
+WORKDIR /usr/src/group-generator
+
+# Sets a HOME environment variable to a directory inside the the container
+ENV HOME=/usr/src/group-generator/data
+
+# Creates the directory for saving the app  state after the program have been run and set permissions
+RUN mkdir -p $HOME && chown -R root:root $HOME
```
The application will automatic handles that already. Use lighter alternative base image for smaller image size

t-desmond replied:
The current image size using this dockerfile is too heavy.. 
This is surely due to the use of ubuntu as base image in the second stage.. A good approach may be to:
- compress the binary executable in the first stage
- use a distroless image like `scratch` in the second stage

micheal-ndoh replied:
> The current image size using this dockerfile is too heavy.. This is surely due to the use of ubuntu as base image in the second stage.. A good approach may be to:
> 
> * compress the binary executable in the first stage
> * use a distroless image like `scratch` in the second stage

OK Thanks @t-desmond  I will  update the ubuntu base image to scratch base image
#7

micheal-ndoh committed:
Fix Dockerfile

Updated Dockerfile to use scratch base image so as to create a much more smaller image for the group -generator application

marcjazz reviewed (changes requested):
Create a new user for the application with limited access and set the `HOME` env value to that of this user.

Marcjazz commented:
@micheal-ndoh Please can you reopen this PR or create a new one that fulfills the same purpose

micheal-ndoh commented:
> @micheal-ndoh Please can you reopen this PR or create a new one that fulfills the same purpose
Sorry about that @Marcjazz  I mistakenly deleted my branch holding the changes made

marcjazz committed:
Merge branch 'master' into docker
